x-pack/plugins/observability_solution/logs_explorer/public/state_machines/data_views/src/services/data_views_service.ts

```diff
@@ -18,22 +24,26 @@ interface LoadDataViewsServiceDeps {
 export const loadDataViews =
   ({ dataViews }: LoadDataViewsServiceDeps): InvokeCreator<DataViewsContext, DataViewsEvent> =>
   async (context) => {
-    const searchParams = context.search;
-
-    if (context.cache.has(searchParams)) {
-      return context.cache.get(searchParams);
+    if (context.cache.has(getSearchCacheKey(context))) {
+      return context.cache.get(getSearchCacheKey(context));
     }
 
     return dataViews
       .getIdsWithTitle()
       .then((views) => views.map(DataViewDescriptor.create))
-      .then((views) => searchDataViews(views, searchParams));
+      .then((views) => searchDataViews(views, context.search, context.filter));
   };
 
-export const searchDataViews = (dataViews: DataViewDescriptor[], search: DataViewsSearchParams) => {
+export const searchDataViews = (
+  dataViews: DataViewDescriptor[],
+  search: DataViewsSearchParams,
+  filter: DataViewsFilterParams
+) => {
   const { name, sortOrder } = search;
+  const { dataType } = filter;
 
   return dataViews
     .filter((dataView) => Boolean(dataView.name?.includes(name ?? '')))
+    .filter((dataView) => !dataType || Boolean(dataView.dataType === dataType))
     .sort(createComparatorByField('name', sortOrder));
 };
```

Comment:
**suggestion**: this might be a good point to do a small refactor and introduce an object parameter for the search criteria, although is not there yet, we might need to filter only one day without passing the search, and having them listed as inline parameters might me more contrived. 
```ts
interface DataViewsSearchCriteria {
  filter?: DataViewsFilterParams;
  search?: DataViewsSearchParams;
}

export const searchDataViews = (
  dataViews: DataViewDescriptor[],
  searchCriteria: DataViewsSearchCriteria
) => {
  const { filter = {}, search = {} } = searchCriteria;
  const { name, sortOrder = 'asc' } = search;
  const { dataType } = filter;

  return dataViews
    .filter((dataView) => Boolean(dataView.name?.includes(name ?? '')))
    .filter((dataView) => !dataType || Boolean(dataView.dataType === dataType))
    .sort(createComparatorByField('name', sortOrder));
};
```